apps/desktop/src/lib/branches/branchListing.ts

```diff
@@ -0,0 +1,82 @@
+			);
+			console.log(branches);
+			return branches;
+		} catch (err: any) {
+			console.error(err);
+		}
+	}
+}
+
+/// Represents a branch that exists for the repository
+/// This also combines the concept of a remote, local and virtual branch in order to provide a unified interface for the UI
+/// Branch entry is not meant to contain all of the data a branch can have (e.g. full commit history, all files and diffs, etc.).
+/// It is intended a summary that can be quickly retrieved and displayed in the UI.
+/// For more detailed information, each branch can be queried individually for it's `BranchData`.
+export class BranchListing {
+	/// The name of the branch (e.g. `main`, `feature/branch`), excluding the remote name
+	name!: string;
+	/// This is a list of remote that this branch can be found on (e.g. `origin`, `upstream` etc.).
+	/// If this branch is a local branch, this list will be empty.
+	remotes!: string[];
+	/// The branch may or may not have a virtual branch associated with it
+	virtualBranch?: VirtualBranchReference | undefined;
+	/// The number of lines added within the branch
+	/// Since the virtual branch, local branch and the remote one can have different number of lines removed,
+	/// the value from the virtual branch (if present) takes the highest precedence,
+	/// followed by the local branch and then the remote branches (taking the max if there are multiple)
+	/// If this branch has a virutal branch, lines_added does NOT include the uncommitted lines.
+	linesAdded!: number;
+	/// The number of lines removed within the branch
+	/// Since the virtual branch, local branch and the remote one can have different number of lines removed,
+	/// the value from the virtual branch (if present) takes the highest precedence,
+	/// followed by the local branch and then the remote branches (taking the max if there are multiple)
+	/// If this branch has a virutal branch, lines_removed does NOT include the uncommitted lines.
+	linesRemoved!: number;
+	/// The number of files that were modified within the branch
+	/// Since the virtual branch, local branch and the remote one can have different number files modified,
+	/// the value from the virtual branch (if present) takes the highest precedence,
+	/// followed by the local branch and then the remote branches (taking the max if there are multiple)
+	numberOfFiles!: number;
+	/// The number of commits associated with a branch
+	/// Since the virtual branch, local branch and the remote one can have different number of commits,
+	/// the value from the virtual branch (if present) takes the highest precedence,
+	/// followed by the local branch and then the remote branches (taking the max if there are multiple)
+	numberOfCommits!: number;
+	/// Timestamp in milliseconds since the branch was last updated.
+	/// This includes any commits, uncommited changes or even updates to the branch metadata (e.g. renaming).
+	updatedAt!: number;
```

Comment:
Since the typing here doesn't indicate it's milliseconds, it's easy to get wrong. I think I have seen somewhere that `_<unit>` suffixes were used to make it painfully clear what is what.